src/components/features/Gallery/Gallery.js

```diff
@@ -13,8 +16,92 @@ import {
 import { faHeart } from '@fortawesome/free-regular-svg-icons';
 import ReactTooltip from 'react-tooltip';
 import StarsRating from '../StarsRating/StarsRating';
+import {
+  getAll,
+  getCompare,
+  toggleProductCompare,
+  toggleFavoriteProduct,
+} from '../../../redux/productsRedux';
+import { getGalleryCategories } from '../../../redux/categoriesRedux';
 
 const Gallery = () => {
+  const dispatch = useDispatch();
+  const fadeTimer = 250;
+  const allProducts = useSelector(getAll);
+  const galleryCategories = useSelector(getGalleryCategories);
+  const getCompareProducts = useSelector(getCompare);
+  const [activatePhotoFade, setActivatePhotoFade] = useState(false);
+  const [activateSliderFade, setActivateSliderFade] = useState(false);
+  const [activeCategory, setActiveCategory] = useState('topSeller');
+  const [productIndex, setProductIndex] = useState(0);
+
+  const categoryProducts = allProducts.filter(
+    item => item.galleryCategory === activeCategory
+  );
+  const [activeProduct, setActiveProduct] = useState(categoryProducts[0].id);
+  const handleCategoryChange = (e, newCategory) => {
+    e.preventDefault();
+    setActivatePhotoFade(true);
+    setActivateSliderFade(true);
+
+    setTimeout(() => {
+      setActiveCategory(newCategory);
+      setActivatePhotoFade(false);
+      setActivateSliderFade(false);
+    }, fadeTimer);
+  };
+
+  useEffect(() => {
+    setActiveProduct(categoryProducts[0].id);
+  }, [activeCategory]);
+
+  const handleProductChange = (e, newProduct) => {
+    e.preventDefault();
+    setActivatePhotoFade(true);
+
+    setTimeout(() => {
+      setActiveProduct(newProduct);
+      setActivatePhotoFade(false);
+    }, fadeTimer);
+  };
+
+  useEffect(() => {
+    setProductIndex(categoryProducts.indexOf(showProduct));
+  }, [activeProduct]);
+
+  const showProduct = allProducts.find(item => item.id === activeProduct);
+
+  const handleCompare = (e, id) => {
+    e.preventDefault();
+    if (getCompareProducts.length < 4) {
+      dispatch(toggleProductCompare(id));
+    }
+  };
+
+  const handleFavorite = (e, id) => {
+    e.preventDefault();
+    dispatch(toggleFavoriteProduct(id));
+  };
+
+  const handleRight = e => {
```

Comment:
Może przydałoby się dodać fade in-out również do sliderów? Bo na chwile obecną jest transition przy klikaniu wszystkiego poza sliderem :)

Reply:
To było trochę celowo zrobione, żeby slider był trochę bardziej responsywny. Do tego w zadaniu nie ma nic na ten temat. Poprawiłem :)